doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,143 @@
+
+- Use a `PARTITION BY` clause on the relevant column to ensure that data with similar values for that column are stored close together.
+- Add a filter to your query that only returns true for a particular range of values in that column.
+
+Filters that consist of arithmetic, date math, and comparisons are generally eligible for pushdown, including all the examples in this page. However, more complex filters might not be. You can check whether the filters in your query can be pushed down by using [the
+`filter_pushdown` option](/sql/explain-plan/#output-modifiers) in an `EXPLAIN` statement.
+
+Some common functions, such as casting from a string to a timestamp, can prevent filter pushdown for a query. For similar functions that _do_ allow pushdown, see [the pushdown functions documentation](/sql/functions/pushdown/).
+
+## Requirements
+
+Materialize imposes some restrictions on the list of columns in the partition-by clause.
+
+- This clause must list a prefix of the columns in the collection. For example, if you're creating a table that partitions by a single column, that column must be the first column in the table's schema definition.
+- Only certain types of columns are supported. This includes:
+    - all fixed-width integer types, including `smallint`, `integer`, and `bigint`;
+    - date and time types, including `date`, `time`, `timestamp`, `timestamptz`, and `mz_timestamp`;
+    - string types like `text` and `bytea`;
+    - `boolean` and `uuid`;
+    - `record` types where all fields types are supported.
+
+We intend to relax some of these restrictions in the future.
+
+## Examples
+
+These examples create real objects. After you have tried the examples, make sure to drop these objects and spin down any resources you may have created.
+
+### Partitioning by timestamp
+
+For timeseries or "event"-type collections, it's often useful to partition the data by timestamp.
+
+1. First, create a table called `events`.
+    ```mzsql
+    -- Create a table of timestamped events. Note that the `event_ts` column is
+    -- first in the column list and in the parition-by clause.
+    CREATE TABLE events (
+        event_ts timestamptz,
+        content text
+    ) WITH (
+        PARTITION BY (event_ts)
+    );
+    ```
+
+1. Insert a few records, one "older" record and one more recent.
+    ```mzsql
+    INSERT INTO events VALUES (now()::timestamp - '1 minute', 'hello');
+    INSERT INTO events VALUES (now(), 'world');
+    ```
+
+1. If we run a select statement against the data sometime in the next couple of minutes, we return one row but not the other.
+    ```mzsql
+    SELECT * FROM events WHERE event_ts > mz_now() - '2 minutes';
+    ```
+
+1. An `EXPLAIN FILTER PUSHDOWN` statement shows that we're able to fetch a subset of the parts in our collection... only those parts that contain data with recent timestamps.
+    ```mzsql
+    EXPLAIN FILTER PUSHDOWN FOR
+    SELECT * FROM events WHERE event_ts + '2 minutes' > mz_now();
+    ```
```

Comment:
Ideally we'd link to `EXPLAIN FILTER PUSHDOWN`, but that's not documented anywhere to my knowledge yet. That may make sense as a separate PR?

Reply:
Actually, it is https://materialize.com/docs/sql/explain-filter-pushdown/